src/client/component/workshop.cpp

```diff
@@ -72,6 +63,113 @@ namespace workshop
 		}
 	}
 
+	std::string get_mod_name(const std::string& mod_id)
+	{
+		if (mod_id == "usermaps")
+		{
+			return mod_id;
+		}
+
+		const utils::nt::library host{};
+		std::string path;
+
+		if (game::is_server())
+		{
+			const auto base_path = host.get_folder().generic_string();
+			path = utils::string::va("%s/mods/%s/zone/workshop.json", base_path.data(), mod_id.data());
+		}
+		else
+		{
+			return mod_id;
+		}
+
+		std::string json_str;
+		utils::io::read_file(path, &json_str);
```

Comment:
there is a shorter read_file that directly returns a string
```suggestion
		std::string json_str = utils::io::read_file(path);
```

Reply:
or you respect the return value of your current read_file call. it returns a bool whether reading was successful or not